wcfsetup/setup/db/install.sql

```diff
@@ -1446,6 +1446,20 @@ CREATE TABLE wcf1_trophy(
 	KEY(categoryID)
 );
 
+DROP TABLE IF EXISTS wcf1_unfurl_url;
+CREATE TABLE wcf1_unfurl_url (
+	urlID INT(10) NOT NULL AUTO_INCREMENT PRIMARY KEY,
+	url TEXT NOT NULL,
+	urlHash VARCHAR(40) NOT NULL UNIQUE KEY (urlHash),
+	title VARCHAR(255) NOT NULL DEFAULT '',
+	description TEXT,
+	imageUrl TEXT NOT NULL DEFAULT '',
+	imageType VARCHAR(255) NOT NULL DEFAULT 'NOIMAGE',
+	imageHash VARCHAR(45) NOT NULL DEFAULT '',
```

Comment:
I dislike this, because it contains both the extension as well as the hash itself. May I suggest the following:

1. Use the `urlHash` as the `imageHash`.
2. Change to column to `storedImageExtension VARCHAR(255) DEFAULT NULL,`

And then using `storedImageExtension` in place of `imageHash`. If it is `null` no image is stored locally.

Reply:
I dislike using the `urlHash` as `imageHash`. 

On many pages (e.g. WoltLab.com) the same image is used as the logo on various pages. If we save the image with the `urlHash` we duplicate the image quite often. If we save the `imageHash`, we store the image only once.

Reply:
Maybe a dedicated database table for the images is in order then? All the `image*` columns should be moved there, with this table only containing the `imageID`. This theoretically would also allow pruning the images independently after a certain time to further save disk space. The `imageType` would then be automatically derived from the information stored in that image table. If `imageID IS NULL`, then no image will be shown.